src/object_hash.rs

```diff
@@ -1,28 +1,85 @@
+use base32;
+use base64;
+use bit_vec::BitVec;
 use error::Result;
 use obj_ser::to_string;
+use ripemd160::Ripemd160;
 use serde::ser::Serialize;
 use sha2::{Digest, Sha256};
 
 pub fn get_base64_hash<T>(object: &T) -> Result<String>
 where
     T: Serialize,
 {
-    use base64;
-
-    // create a Sha256 object
-    let mut hasher = Sha256::default();
-
-    // write input message
-    hasher.input(&to_string(object)?.as_bytes());
-
-    Ok(base64::encode(&hasher.result()))
+    Ok(base64::encode(
+        &Sha256::digest(&to_string(object)?.as_bytes()),
+    ))
 }
 
-pub fn get_chash<T>(_object: &T) -> Result<String>
+pub fn get_chash<T>(object: &T) -> Result<String>
 where
     T: Serialize,
 {
-    unimplemented!()
+    let hash = Ripemd160::digest(&to_string(object)?.as_bytes());
```

Comment:
wow! is there any available crate that does the same thing? I'm surprised that you implement the whole thing yourself! It will be a problem if there are bugs in this code in future, so I really want to use a third party code which is widely accepted.